healthcheck/healthcheck.go

```diff
@@ -127,11 +130,36 @@ func checkBackend(currentBackend *BackendHealthCheck) {
 	}
 }
 
+func (backend *BackendHealthCheck) newRequest(serverURL *url.URL) *http.Request {
+	// make a copy of the url
+	u := &url.URL{}
+	*u = *serverURL
+
+	if backend.Options.Port != 0 {
+		u.Host = net.JoinHostPort(u.Hostname(), strconv.Itoa(backend.Options.Port))
+	}
+	u.Path = u.Path + backend.Path
+
+	req := &http.Request{
```

Comment:
(I saw you made the change.)

Yeah it seems a bit cumbersome; personally though, I'd like to err on the side of safety and stability.